src/main/cdp/capsolver-chrome-extension.ts

```diff
@@ -0,0 +1,32 @@
+import * as fs from 'fs/promises';
+
+import { ConsoleLogger } from '../logger';
+
+
+export class CapsolverChromeExtension {
+    protected logger = new ConsoleLogger();
+
+    constructor(protected extensionPath: string, protected apiKey: string) {}
+
+    async addApiKey() {
+        await this.modifyApiKey(this.apiKey);
```

Comment:
Was wondering if it'd be easier to use the same `@config` approach that we use everywhere? Or is this not feasible because it has to be done before Chrome is run (i.e. outside of engine lifecycle)?

Reply:
I first started implementing @config approach but then realised it would actually be more complicated to instantiate the class with inversify and 1 injected constructor param, so I just simplified so that inversify is not needed at all.

Reply:
I understand that DI is oftentimes doesn't seem necessary, however, cases like these then become "exceptions" in the architecture, which complicate things when it comes to maintenance, extending, shifting things around.

I mean imagine something as simple as adding a log line — not an unreasonable possible requirement — and surely it's possible to just pass `logger` as a parameter where it's needed (it's way simpler than refactoring). Add a few steps in the same direction and we'll end up with another pre-framework Service API.

Reply:
I apologize for not providing the full context that would help you understand my implementation decision. I encapsulated the portion related to CapSolver specifics in a separate class, placing it in the same directory as the ChromeLauncher class. Initially, I assumed that the "@config" approach was used universally, but I later realized the following:

- CDP doesn't utilize Dependency Injection (DI).
- We set the path to the extension from both the worker and the drone. Since we already have the configuration in those places, it makes sense to send the API key as well.

I'm sorry if my initial comment suggested that my decision was solely based on implementing the easiest solution. I am genuinely concerned about the architecture and maintaining consistent standards across the codebase. Therefore, ease of implementation is not the primary factor in this case.

Reply:
Thanks for the additional context — all makes sense 👍